app/src/main/java/io/gnosis/kouban/ui/transaction/TransactionsViewModel.kt

```diff
@@ -58,6 +61,16 @@ class TransactionsViewModel(
                 emit(ENSName(it))
             }
         }
+
+    private fun withDateLabels(transactions: List<Transaction>): List<Any> =
+        mutableListOf<Any>().apply {
```

Comment:
you. might be able to do this with the `fold` operation ;)

Reply:
Challenge accepted :)